### PR TITLE
Fix perception.velocity direction bug and perception frequency bug

### PR DESCRIPTION
**Pull request by yuntianyi-chen**
When using the _garage_perception.bash_ script to register the obstacles, the green arrow of obstacle velocity direction doesn't change in the Dreamview because the `perception.velocity` is not updated in the `linear_project_perception` function. Here we add a line of code to update the `perception.velocity`.

### CODE REVIEW

**Comment by yuntianyi-chen** [comment]
There are also some bugs in the perception publishing frequency.

The default value of period 0.1 would result in the error of publish delay. We could change it to 10. Therefore the sleep_time should be float and equal to 0.1 and _s_delta_t should equal to sleep_time.

Now we can publish the perception normally.

**Comment by YuqiHuai** [comment]
I agree with the changes proposed.
1. We should not use `int()` for `sleep_time` because 10 Hz would be publishing 10 times per second, and the corresponding `sleep_time` should be `0.1`. If `int()` is used, `sleep_time = int(0.1)` will assign `0` to `sleep_time`, which will keep publishing obstacle detection without any sleep time in between.
2. The current default value of `period = 0.1` will cause `sleep_time = int(1.0 / 0.1)` which is 10 seconds between messages, or `# 0.1 Hz`.
2. `_s_delta_t` should always be equal to `sleep_time` because this variable is used to calculate the displacement of obstacles during the time publisher is "sleeping". 

With the changes accepted, we can further improve the description of command line argument `-p` to "publish frequency" and its default value of 10 indicates 10 Hz.

**Review by YuqiHuai — APPROVED** [review]
1. Direction of velocity is correctly updated.
2. sleep_time is float instead of int. Sleep time should be about 0.1 seconds, if `int` is used, this value will never be 0.1 (e.g., `int(0.1) == 0`)
3. `_s_delta_t`should be the same as `sleep_time`, it is used to calculate the displacement of obstacles.

**Comment by YuqiHuai** [comment]
Hi @daohu527 ! I am wondering if anyone can help merge this fix. I noticed Apollo 8.0 has some documentation with using SimControl and so far this `replay_perception.py` script is essential to feed simulated obstacles into `/apollo/perception/obstacles` so we can use SimControl to create more interesting scenarios.
Without this fix, `replay_perception.py` is essentially only updating the location of obstacles at an incorrect rate.

**Comment by daohu527** [comment]
This script may need further optimization, we cannot assume a constant velocity

File: modules/tools/perception/replay_perception.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def linear_project_perception(description, prev_perception):
                 get_point(trace[i - 1], trace[i], ratio))
             perception.theta = math.atan2(trace[i][1] - trace[i - 1][1],
                                           trace[i][0] - trace[i - 1][0])
-
+            perception.velocity.CopyFrom(get_velocity(perception.theta, description["speed"]))
```
[the remark this answers is not in the document]
Comment (YuqiHuai) [reply modules/tools/perception/replay_perception.py:234]: Currently what we have observed is, although the obstacle is moving along the specified trajectory, its velocity's direction remains as the initial velocity's direction. As an example, consider an obstacle making a right turn (90 degrees). `replay_perception` will be able to update the obstacle's 2D bounding box and its heading, but does not update its direction of velocity. You will be observing a vehicle whose velocity is 90 degrees from the vehicle's heading. Therefore the velocity needs to be recalculated when the obstacle's heading changes, assuming obstacle is always at constant velocity.

**Comment by YuqiHuai** [comment]
I agree with you that the script needs further optimization due to its current assumption of obstacle traveling at constant velocity. I thought this was intended as it is using the most simplistic obstacle kinematic. As a result, an obstacle may be making unrealistic turns at any specific speed.

I am actively trying to solve this issue by implementing a new `replay_perception.py` that considers the speed limit of the lane in which the obstacle is traveling. The limitation of this approach is the obstacle will always be moving without stopping for 1) traffic control devices, and 2) other road traffic participants. Do you have any further suggestions or comments with regard to this proposal?

File: modules/tools/perception/replay_perception.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (daohu527) [reply modules/tools/perception/replay_perception.py:234]: I know the velocity needs to be recalculated, but the position limit should be accepted. Of course, this modification can be accept first, and then optimize

[the remark this answers is not in the document]
Comment (YuqiHuai) [reply modules/tools/perception/replay_perception.py:234]: What do you mean by "the position limit should be accepted"? and yes it sounds good to take this as a fix to an existing problem and we can try to fix it later.

File: modules/tools/perception/replay_perception.py [hunk 2]
```diff
@@ -272,9 +272,9 @@ def perception_publisher(perception_channel, files, period):
     node = cyber.Node("perception")
     writer = node.create_writer(perception_channel, PerceptionObstacles)
     perception_description = load_descrptions(files)
-    sleep_time = int(1.0 / period)  # 10Hz
+    sleep_time = float(1.0 / period)  # 10Hz
```
[the remark this answers is not in the document]
Comment (yuntianyi-chen) [reply modules/tools/perception/replay_perception.py]: Fix and committed. @daohu527

**Comment by daohu527** [comment]
@yuntianyi-chen Pls rebase the commit, format like "fix(tools): xxx". 

The ci is broken, we will fix it later.

**Comment by yuntianyi-chen** [comment]
@daohu527 I Rebased last three commits and changed title.

**Review by daohu527 — APPROVED** [review]
Improve the velocity. currently it is a constant speed, can be changed to calculate according to the position

**Pull request merged**